cmd/utils.go

```diff
@@ -283,27 +283,48 @@ func getProjectDir(config *RootCommandConfig) (string, error) {
 
 // IsValidProjectName tests the given string against Appsody name rules.
 // This common set of name rules for Appsody must comply to Kubernetes
-// resource and Docker container name rules. The current rules are:
+// resource name, Kubernetes label value, and Docker container name rules.
+// The current rules are:
 // 1. Must start with a lowercase letter
 // 2. Must contain only lowercase letters, digits, and dashes
 // 3. Must end with a letter or digit
-// 4. Must be less than 128 characters
+// 4. Must be 68 characters or less
 func IsValidProjectName(name string) (bool, error) {
-	match, err := regexp.MatchString("^[a-z]([a-z0-9-]*[a-z0-9])?$", name)
+	if name == "" {
+		return false, errors.New("Invalid project-name. The name cannot be an empty string")
+	}
+	if len(name) > 68 {
+		return false, errors.Errorf("Invalid project-name \"%s\". The name must be 68 characters or less", name)
+	}
 
+	match, err := regexp.MatchString("^[a-z]([a-z0-9-]*[a-z0-9])?$", name)
 	if err != nil {
 		return false, err
 	}
 
 	if match {
-		if len(name) < 128 {
-			return match, nil
-		}
-		return false, errors.Errorf("Invalid project-name \"%s\". The name must be less than 128 characters", name)
+		return true, nil
+	}
+	return false, errors.Errorf("Invalid project-name \"%s\". The name must start with a lowercase letter, contain only lowercase letters, numbers, or dashes, and cannot end in a dash.", name)
+}
+
+func IsValidKubernetesLabelValue(value string) (bool, error) {
+	if value == "" {
+		return true, nil
+	}
+	if len(value) > 68 {
+		return false, errors.New("The label must be less than 128 characters")
 	}
 
-	return match, errors.Errorf("Invalid project-name \"%s\". The name must start with a lowercase letter, contain only lowercase letters, numbers, or dashes, and cannot end in a dash.", name)
+	match, err := regexp.MatchString("^[a-z0-9A-Z]([a-z0-9A-Z-_\\.]*[a-z0-9A-Z])?$", value)
```

Comment:
If I am reading this correctly, we are letting the value have `\` and `_` - is this allowed my kubernetes?

Reply:
`_` is allowed but not `\`

The `\\` is to escape a single `\` in go, so that it can escape the `.` in regex.

However, I believe escaping the `.` in regex is optional when in `[ ]` so I've removed the slashes.